streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignorConfiguration.java

```diff
@@ -47,59 +47,21 @@
 
     private final String logPrefix;
     private final Logger log;
-    private final AssignmentConfigs assignmentConfigs;
-    @SuppressWarnings("deprecation")
-    private final org.apache.kafka.streams.processor.PartitionGrouper partitionGrouper;
-    private final String userEndPoint;
     private final TaskManager taskManager;
-    private final StreamsMetadataState streamsMetadataState;
```

Comment:
Sorry for all the lines of changes here, but the inconsistent method signatures and object construction was making this class hard to follow. All I'm doing here is making the methods conform to the same style, and moving the construction of any config that isn't needed elsewhere to its getter method. This allows us to remove a lot of these class variables

Reply:
Do we know if the corresponding "getters" are called often or not? I guess the main idea behind having those member variables was to "parse" the config once as it's immutable anyway instead of each time a getter is called?

Reply:
They are not. We construct the `AssignorConfiguration` object once in `StreamsPartitionAssignor#configure` and call the getters immediately afterwards, and only there